### ISSUE TITLE
InvalidSignatureExpection - Signature does not match

### ISSUE THREAD

**Issue by paul-doherty** [aws-amplify/amplify-js#3270]
**Describe the bug**
[React Native in release mode]
Signing into the application. Sign out of the application. Force closing the application and again sign into the application. This results in the following rejection by the API gateway.

```The Canonical String for this request should have been 'POST /content accept:*/* content-type:application/json; charset=utf-8 host:mobile-content<CUT_FOR_BREVITY> x-amz-date:20190513T160136Z x-amz-security-token:nJg9....<CUT_FOR_BREVITY> accept;content-type;host;x-amz-date;x-amz-security-token 91a4...<CUT_FOR_BREVITY>' The String-to-Sign should have been 'AWS4-HMAC-SHA256 20190513T160136Z 20190513/eu-west-1/execute-api/aws4_request bc2d16a480fec2ec7c0c97d614399f20087e3db1484255487478de45460995a8'```

I struggle to reproduce this in debug mode but happens consistently in release but that could just be me.

**To Reproduce**

- Use Auth.signIn to sign into the application
- -  Everything works as expected
- Use Auth.signOut to sign out from the application
- Force close the application
- Open the application and sign in again. All requests now fail. 
- Again force close the app
- Again open the app. Signing works again and everything functions as expected

**Expected behavior**
All request should be signed correctly.

**Smartphone (please complete the following information):**
 - OS: Android
 - Version 8 and 9

**Additional context**
```05-13 17:06:04.056: I/ReactNativeJS(26377): '[ERROR] 06:04.54 AWSPinpointProvider - record event failed. ', [InvalidSignatureException: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.
05-13 17:06:04.056: I/ReactNativeJS(26377): The Canonical String for this request should have been
05-13 17:06:04.056: I/ReactNativeJS(26377): 'POST
05-13 17:06:04.056: I/ReactNativeJS(26377): /v1/apps/dc4c....<CUT>/events/legacy
05-13 17:06:04.056: I/ReactNativeJS(26377): host:pinpoint.us-east-1.amazonaws.com
05-13 17:06:04.056: I/ReactNativeJS(26377): x-amz-content-sha256:3d246bbb92b500e412ab1df6e0401c45f3e7ec1b207d7f152c9f0947926c47a2
05-13 17:06:04.056: I/ReactNativeJS(26377): x-amz-date:20190513T160603Z
05-13 17:06:04.056: I/ReactNativeJS(26377): x-amz-security-token:AgoGb3JpZ2luEBwaCWV1LXdlc3QtMSKAAg1hRBFRNoS3tB+VjTruGZj1mzOMWy4xeQdYwn/E8a7dmHCq2zFwkczy0p+Rf89ORPh8gV4RiHI88xC3MaaOvXKZWaSn5jh470nMVQX5tBtAZk6joYO2+moU2BO5MpVw1hew9gXzWlgN/xBpoy7gBHiFIMo/4BkPImTPX65xO0bGJsyjFARczv7CTWC2YtUTfVygRyoVM6Le1if6RltCNaqUdJvLuTbRyyZOo8dv83DDvcjUm+wuLAKDvXc9l2N22D8AkNIdjIQTZOdMlzZOIQpguUho9FByKTVK76GINog02h5Rup3crMDeibTn6tT2IauPa3M7r2Hdg+IAB1FNwOIqtQYIwf//////////ARAAGgw5MDE1Njg1MTI0ODgiDBW4EOPNhSURS0M3HyqJBqDdYzXll+1vV+N+sN+ALJSwSR6woeKY40gaFzfUyV3CtCmylmAc7I16c/T3IcPt1rOzEnyXvPjqxh9rhMAmIz4T2ZhOHNN50duLce6goWWHXRcTgzFThjGSx0MteVRC71odTRJJCoemH0xuXUE0uZmTFaEInD1QAPTGzVBO+fbyH5KjBTkWRgizNuoY1mJGIIkTgm7t6EevMsJE4ObT4/CjOIEbszBX0k9i9/7P5k3nm6vdjp1+TZ8sxggBoK0STiUyNKfQexigDwnve3AgzYhNu5A48g4eYsMh8s+ed4L6i+CDv+PnNtm0P8QFU6ZdHwxXA9Ej3vyorWdtxPz3kSeE8ltOma2qrZLOXQrGGBu9EeEE17XPwhzl67lEesCMrgIgggtATjhkghp8RNLjyVRNXvLS0l7OABV66a63wWUbYduShab4B05LYpfnJMdgiC9DO7krYDP3W7zGaulVj/gVW2ASGBg1PNIPIUjKJg/3aDdNVbShtL216tJHJuLghPXL268KHW/bWJUubqSxqJSireL9l91piMEelt2LJNsJurj9NucCgHV5WM4wY8nrfUjkl437TgWekDrAA8UN46AqnwfHNXUs0Kn/0ZbVgwehZ2dezPwPCKBujo560urGwP2Iti8lHDZDA9mkrzp8bIIg+vPfnkSRB+epYuPgGUpzMCHJuswYuI7RzpJdNBGMWJQiTVJrKOpVG9XY0jfvd092YQiuDcHx8TLl4jQOPyK870GtuLQMsqolJ0S6d8jZq9u6eNDmrXy12u6IXKMpGi9GCvG/e3W/GaoDu3AJXKmTd2tQgYV7cic7a0rOqT67rwotbAu+4iAV29/26Zya2tsx1qua/NmeqeDciQxk6uRNybckFFnkVtaV+y6eFqSu+AL0JyVXGW6pauwwN4F+P6QHEEW29pb3pAhSp43RNvu3abqiSKotT/EwXCUE/4S5gRj4dvA0R0JBRS7E3tKfIcBnniLlC8odLXlksww6SQLc86J+PWRXREt7TYRycVxqZZu//5DBDKADyjDiq+bmBQ==
05-13 17:06:04.056: I/ReactNativeJS(26377): x-amz-user-agent:aws-amplify/1.0.25 react-native aws-amplify/1.0.25 react-native callback
05-13 17:06:04.056: I/ReactNativeJS(26377): host;x-amz-content-sha256;x-amz-date;x-amz-security-token;x-amz-user-agent
05-13 17:06:04.056: I/ReactNativeJS(26377): 636d6bcfea9e00262f829c83e341b6cbd6b00d5693d38b0635c64db016831201'
05-13 17:06:04.056: I/ReactNativeJS(26377): The String-to-Sign should have been
05-13 17:06:04.056: I/ReactNativeJS(26377): 'AWS4-HMAC-SHA256
05-13 17:06:04.056: I/ReactNativeJS(26377): 20190513T160603Z
05-13 17:06:04.056: I/ReactNativeJS(26377): 20190513/us-east-1/mobiletargeting/aws4_request
05-13 17:06:04.056: I/ReactNativeJS(26377): 40ac41a0397051b07a09c48fe4b5b6391e47bd895ea18fcb65faa95b1527455b'
05-13 17:06:04.056: I/ReactNativeJS(26377): ]```

***You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app.***

**Comment by elorzafe** [comment]
@paul-doherty can you enable `Amplify.Logger.LOG_LEVEL = 'DEBUG'` on your app, maybe we can find more info.

**Comment by paul-doherty** [comment]
Struggled to get the logs as I couldn't recreate it in release mode. What I eventually found out was before the user logs in we use `API.get` on an unprotected gateway. Afterwards the user logs in and Amplify appears to still use the unauth credentials. I changed to use fetch directly one that call and the problem went away.

I presumed, perhaps incorrectly, that signing in with Auth.signIn would clear any existing credentials and use the new ones for signing requests. This doesn't appear to be the case
